app/importers/file_importers/base_csv_importer.rb

```diff
@@ -0,0 +1,37 @@
+class BaseCsvImporter < Struct.new :school_scope, :client, :log, :progress_bar, :import_record_detail
+  def import
+    import_record_detail.start
+    before_import if self.respond_to?(:before_import)
+
+    @data = CsvDownloader.new(
+      log: log, remote_file_name: remote_file_name, client: client, transformer: data_transformer
+    ).get_data
+
+    @data.each.each_with_index do |row, index|
+      if filter.include?(row)
+        import_record_detail.log_processed
+        import_row(row)
+      else
+        import_record_detail.log_excluded
+      end
+      ProgressBar.new(log, remote_file_name, @data.size, index + 1).print if progress_bar
+    end
+    after_import if self.respond_to?(:after_import)
+    import_record_detail.complete
+  end
+
+  def data_transformer
+    CsvTransformer.new
+  end
+
+  def filter
+    SchoolFilter.new(school_scope)
+  end
+
+  # Mirror import_record_detail functionality to simplify importers
+  def method_missing(name, *args, &block)
```

Comment:
Let's avoid using `method_missing`, it's not easy to read and makes the code harder to debug. Check out the ruby style guide comments: https://github.com/bbatsov/ruby-style-guide#user-content-no-method-missing

Reply:
Yeah, so many strange features of Ruby. I'll switch use explicit delegation.